src/services/query-options.js

```diff
@@ -0,0 +1,248 @@
+    const include = new QueryBuilder().getIncludes(this._model, fields.length ? fields : null);
+    return include.length ? include : null;
+  }
+
+  /**
+   * @param {sequelize.model} model Sequelize model that should be targeted
+   * @param {boolean} options.includeRelations Include BelongsTo and HasOne relations by default
+   * @param {boolean} options.tableAlias Alias that will be used for this table on the final query
+   *  This should have been handled by sequelize but it was needed in order to use
+   *  sequelize.fn('lower', sequelize.col('<alias>.col')) in the search-builder with
+   *  has-many-getter.
+   */
+  constructor(model, options = {}) {
+    this._Sequelize = model.sequelize.constructor;
+    this._schema = Schemas.schemas[model.name];
+    this._model = model.unscoped();
+    this._options = options;
+
+    // Used to compute relations that will go in the final 'include'
+    this._requestedFields = new Set();
+    this._neededFields = new Set();
+    this._scopes = []; // @see sequelizeScopes getter
+
+    // Other sequelize params
+    this._where = [];
+    this._order = [];
+    this._offset = undefined;
+    this._limit = undefined;
+
+    if (this._options.includeRelations) {
+      _.values(this._model.associations)
+        .filter((association) => ['HasOne', 'BelongsTo'].includes(association.associationType))
+        .forEach((association) => this._requestedFields.add(association.associationAccessor));
+    }
+  }
+
+  /**
+   * Add the required includes from a list of field names.
+   * @param {string[]} fields Fields of HasOne and BelongTo relations are
+   *  accepted (ie. 'book.name').
+   */
+  async requireFields(fields) {
+    if (fields) {
+      fields.forEach(this._requestedFields.add, this._requestedFields);
+    }
+  }
+
+  /**
+   * Filter resulting query set with packed primary ids.
+   * This works both for normal collection, and those which use composite primary keys.
+   * @param {string[]} recordIds Packed record ids
+   */
+  async filterByIds(recordIds) {
+    this._where.push(new PrimaryKeysManager(this._model).getRecordsConditions(recordIds));
+  }
+
+  /**
+   * Apply condition tree to those query options (scopes, user filters, charts, ...)
+   * @param {*} filters standard forest filters
+   * @param {*} timezone timezone of the user (required if filtering on dates)
+   */
+  async filterByConditionTree(filters, timezone) {
+    const filterParser = new FiltersParser(this._schema, timezone, { Sequelize: this._Sequelize });
+
+    if (filters) {
+      const whereClause = await filterParser.perform(filters);
+      this._where.push(whereClause);
+
+      const associations = await filterParser.getAssociations(filters);
+      associations.forEach(this._neededFields.add, this._neededFields);
+    }
+  }
+
+  /**
+   * Retrict rows to those matching a search string
+   * @param {string} search search string
+   * @param {boolean|string} searchExtended if truthy, enable search in relations
```

Comment:
Keep to boolean. Don't encourage people to pass a strring while it's expecting a boolean

Reply:
I'm passing the query param directly from the resources-getter.
I'll change the jsdoc